internal/bzlmod/go_deps.bzl

```diff
@@ -57,8 +58,25 @@ def _go_deps_impl(module_ctx):
 
         additional_module_tags = []
         for from_file_tag in module.tags.from_file:
+
+            # We will load all sums from transitively resolved `go.sum` files. We will
+            # only use these after version resolution is completed.
+            for item, new_sum in sums_from_go_mod(module_ctx, from_file_tag.go_mod).items():
+                if item in sums and new_sum != sums[item]:
+                    fail("Multiple mismatching sums for {}@{} found.".format(module_tag.path, sum_version))
+                sums[item] = new_sum
+
             additional_module_tags += deps_from_go_mod(module_ctx, from_file_tag.go_mod)
 
+        # Load sums from all manually specified modules separately.
+        # Fail on duplicates with mismatched sums.
+        for module_tag in module.tags.module:
+            sum_version = _canonicalize_raw_version(module_tag.version)
+            entry = (module_tag.path, sum_version)
```

Comment:
Nit: If `item` and `entry` refer to the same kind of tuple that I think they do, use the same name in both places.